spec/rack/profiler_spec.rb

```diff
@@ -193,6 +194,27 @@ def trigger_dummy_event(name, payload = {})
           parsed_body['events'].map { |e| e['name'] }
         ).to include('rack-profiler.total_time', 'rack-profiler.step')
       end
+
+      context "when authorization is configured" do
+        before do
+          profiler.authorize { |env| env['rack-profiler-allowed'] }
+        end
+
+        it "returns the original response if the request is not authorized" do
+          response = profiler.call(env_with_param)
+          expect(response).to eq(
+            [200, { 'X-My-Header' => 'foo' }, ['hello hello']]
+          )
+        end
+
+        it "returns the profiler results if the request is authorized" do
+          status, headers, body = profiler.call(
+            env_with_param.merge('rack-profiler-allowed' => true)
+          )
+          parsed_body = JSON.parse(body.join)
+          expect(parsed_body).to have_key('events')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.